SEDaily-IOS/GeneralCollectionViewController.swift

```diff
@@ -18,281 +16,144 @@ class GeneralCollectionViewController: UICollectionViewController {
         return SkeletonCollectionView(frame: self.collectionView!.frame)
     }()
     
-    var type: String = ""
-    var tabTitle = ""
-    var tagId = -1
-    var token: NotificationToken?
-    var data: Results<PodcastModel>!
+    var type: PodcastTypes
+    var tabTitle: String
+    var tags: [Int]
+    var categories: [Int]
     
-    var itemCount = 0
+    // ViewModelController
+    private let podcastViewModelController = PodcastViewModelController()
     
-    // MARK: - Paging
-    var loading = false
-    
-    let pageSize = 10
-    let preloadMargin = 5
-    var lastLoadedPage = 0
-    
-    enum APICheckDates {
-        static let newFeedLastCheck = "newFeed"
-    }
-    
-    init(collectionViewLayout layout: UICollectionViewLayout, tagId: Int = -1, type: String) {
-        super.init(collectionViewLayout: layout)
-        
-        self.tagId = tagId
+    init(collectionViewLayout layout: UICollectionViewLayout,
+         tags: [Int] = [],
+         categories: [PodcastCategoryIds] = [],
+         type: PodcastTypes = .new,
+         tabTitle: String = "") {
+        self.tabTitle = tabTitle
         self.type = type
+        self.tags = tags
+        self.categories = categories.flatMap { $0.rawValue }
+        
+        super.init(collectionViewLayout: layout)
     }
     
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-    
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
+
         // Uncomment the following line to preserve selection between presentations
         // self.clearsSelectionOnViewWillAppear = false
-        
+
         // Register cell classes
         self.collectionView!.register(PodcastCell.self, forCellWithReuseIdentifier: reuseIdentifier)
```

Comment:
Consider not using forced unwrap on variables and try to either use the optional or if let/guard let.

Reply:
Could have a guard but this seems to be general practice. Mostly because in a collectionViewController there should always be a collection view. I may be wrong but this is what I see in most Apple projects.

Reply:
You are right that the collectionviewcontroller should have a collection view, but in general practice (at least from my own team coding standard) it's safer to never force unwrap.  Apple projects aren't exactly the best examples of coding practices ;) 

In this case, there is an inconsistency because we use self.collectionview?. addSubview a few lines later.

This is just a suggestion, you don't have to change it.

Reply:
I'm sure it would be caught when we add Swift Lint here in a bit. Haha, I agree that Apple Projects aren't the best coding practices sometimes.

I did change it to `collectionView?` for consistency but I'm more voicing that I'm more 🤷 about tiny things like this overall. Not attacking the way you do things, that's just how I approach the project. 

This of course all gets fixed immediately after adding a linter.